proxmoxtf/resource/file.go

```diff
@@ -638,22 +664,27 @@ func fileGetContentType(d *schema.ResourceData) (*string, diag.Diagnostics) {
 			mkResourceVirtualEnvironmentFileSourceRaw,
 		)
 	}
-
 	if contentType == "" {
 		if strings.HasSuffix(sourceFilePath, ".tar.gz") ||
 			strings.HasSuffix(sourceFilePath, ".tar.xz") {
 			contentType = "vztmpl"
+			// For Proxmox VE 8.4 and later, we can import VM images to  the "import" content type.
+		} else if releaseMajor >= 8 && releaseMinor > 4 && (strings.HasSuffix(sourceFilePath, ".qcow2") ||
+			strings.HasSuffix(sourceFilePath, ".raw") ||
+			strings.HasSuffix(sourceFilePath, ".vmdk")) {
+			contentType = "import"
 		} else {
 			ext := strings.TrimLeft(strings.ToLower(filepath.Ext(sourceFilePath)), ".")
 
 			switch ext {
-			case "img", "iso":
+			case "iso":
```

Comment:
same here, not sure why `img` was removed. I think we should keep this code for backward compatibility.